src/main/scala/is/hail/methods/FilterAlleles.scala

```diff
@@ -171,17 +171,25 @@ object FilterAlleles {
 
     val newRDD2: OrderedRVD =
       if (leftAligned) {
-        OrderedRVD(newMatrixType.orvdType,
-          vsm.rvd.partitioner,
-          filter(vsm.rvd, removeLeftAligned = false, removeMoving = false, verifyLeftAligned = true))
+        filter(
+          vsm.rvd,
+          removeLeftAligned = false,
+          removeMoving = false,
+          verifyLeftAligned = true)
       } else {
-        val leftAlignedVariants = OrderedRVD(newMatrixType.orvdType,
-          vsm.rvd.partitioner,
-          filter(vsm.rvd, removeLeftAligned = false, removeMoving = true, verifyLeftAligned = false))
-
-        val movingVariants = OrderedRVD.shuffle(newMatrixType.orvdType,
-          vsm.rvd.partitioner,
-          filter(vsm.rvd, removeLeftAligned = true, removeMoving = false, verifyLeftAligned = false))
+        val leftAlignedVariants =
+          filter(
+            vsm.rvd,
+            removeLeftAligned = false,
+            removeMoving = true,
+            verifyLeftAligned = false)
+
+        val movingVariants =
+          filter(
+            vsm.rvd,
+            removeLeftAligned = true,
+            removeMoving = false,
```

Comment:
I'm confused how these can be written symmetrically.  One needs a shuffle and the other doesn't.

Reply:
Definitely a mistake, I didn't preserve the shuffle. Will fix.

What's the difference between coerce and shuffle? Is shuffle just a fast-path (skip checking the partition keys) to the slowest (but most complete) type of coercion?